src/resources/glossary.md

```diff
@@ -0,0 +1,19 @@
+---
+title: Glossary
+description: A glossary reference for terminology used across dart.dev
+sitemap: false
+---
+
+The following are definitions of terminology used across the Dart documentation.
+
+## Irrefutable patterns
+
+_Irrefutable patterns_ are patterns that always match. 
+Irrefutable patterns appear in _irrefutable contexts_: 
+_declaration context_ and _assignment context_.
+
+## Refutable patterns
+
+A _refutable pattern_ is a pattern that can be tested against a value to
+determine if the pattern matches the value. If not, the pattern _refutes_ the value.
```

Comment:
It doesn't necessarily need to be addressed now, but defining "Refutable patterns" with "refute" seems a bit tricky to me. I'm not sure most readers will know what refute means.